scripts/apps/dashboard/user-activity/components/UserActivityWidget.tsx

```diff
@@ -0,0 +1,411 @@
+                                preview={(item: IArticle) => {
+                                    this.services.superdesk.intent(
+                                        'preview',
+                                        'item',
+                                        item,
+                                    );
+                                }}
+                                select={angular.noop}
+                                edit={(item: IArticle) => {
+                                    this.services.authoringWorkspace.edit(item);
+                                }}
+                                itemIds={data.itemIds}
+                                itemsById={data.itemsById}
+                                loading={false}
+                            />
+                        </div>
+                    )}
+                </div>
+            </>
+        );
+    }
+
+    render() {
+        const {loading} = this.state;
+
+        return (
+            <div className="widget-container">
+                <div className="main-list" style={{top: 0}}>
+                    <div className="widget-header">
+                        <button
+                            className="widget-header__search-button"
+                            onClick={() =>
+                                this.setState({
+                                    searchOpen: !this.state.searchOpen,
+                                })
+                            }
+                        >
+                            <i className="icon-search" />
+                        </button>
+                        <div className="widget-title">
+                            {gettext('User Activity')}
+                        </div>
+                    </div>
+                    <div
+                        className={`search-box search-box--no-shadow search-box--fluid-height ${
+                            this.state.searchOpen ? '' : 'search-box--hidden'
+                        }`}
+                    >
+                        <form className="search-box__content">
+                            <SelectUser
+                                selectedUserId={this.state.user?._id}
+                                focus={false}
+                                onSelect={(user) => {
+                                    this.setState(
+                                        {user, groups: GET_GROUPS(user._id, this.services), loading: true},
+                                        () => {
+                                            this.fetchGroupsData();
+                                        },
+                                    );
+                                }}
+                            />
+                        </form>
+                    </div>
+                    {
+                        loading && <div className="item-group__loading" />
+                    }
+                    {this.state.user && this.state.groupsData && (
```

Comment:
You shouldn't display old data while new data is loading and a different user is already selected.